src/server.ts

```diff
@@ -0,0 +1,224 @@
+import { EventEmitter } from 'events'
+import * as IlpPacket from 'ilp-packet'
+import * as ILDCP from 'ilp-protocol-ildcp'
+import createLogger from 'ilp-logger'
+import * as cryptoHelper from './crypto'
+import { Connection, ConnectionOpts } from './connection'
+import { ServerConnectionPool } from './pool'
+import { Plugin } from './util/plugin-interface'
+
+const CONNECTION_ID_REGEX = /^[a-zA-Z0-9~_-]+$/
+
+export interface ServerOpts extends ConnectionOpts {
+  serverSecret?: Buffer
+}
+
+/**
+ * STREAM Server that can listen on an account and handle multiple incoming [`Connection`s]{@link Connection}.
+ * Note: the connections this refers to are over ILP, not over the Internet.
+ *
+ * The Server operator should give a unique address and secret (generated by calling
+ * [`generateAddressAndSecret`]{@link generateAddressAndSecret}) to each client that it expects to connect.
+ *
+ * The Server will emit a `'connection'` event when the first packet is received for a specific Connection.
+ */
+export class Server extends EventEmitter {
+  protected serverSecret: Buffer
+  protected plugin: Plugin
+  protected serverAccount: string
+  protected serverAssetCode: string
+  protected serverAssetScale: number
+  protected log: any
+  protected enablePadding?: boolean
+  protected connected: boolean
+  protected connectionOpts: ConnectionOpts
+  private pool: ServerConnectionPool
+
+  constructor (opts: ServerOpts) {
+    super()
+    this.serverSecret = opts.serverSecret || cryptoHelper.randomBytes(32)
+    this.plugin = opts.plugin
+    this.log = createLogger('ilp-protocol-stream:Server')
+    this.connectionOpts = Object.assign({}, opts, {
+      serverSecret: undefined
+    }) as ConnectionOpts
+    this.connected = false
+  }
+
+  /**
+   * Event fired when a new [`Connection`]{@link Connection} is received
+   * @event connection
+   * @type {Connection}
+   */
+
+  /**
+   * Connect the plugin and start listening for incoming connections.
+   *
+   * When a new connection is accepted, the server will emit the "connection" event.
+   *
+   * @fires connection
+   */
+  async listen (): Promise<void> {
+    if (this.connected && this.plugin.isConnected()) {
+      return
+    }
+    this.plugin.registerDataHandler(this.handleData.bind(this))
+    await this.plugin.connect()
+    const { clientAddress, assetCode, assetScale } = await ILDCP.fetch(this.plugin.sendData.bind(this.plugin))
+    this.serverAccount = clientAddress
+    this.serverAssetCode = assetCode
+    this.serverAssetScale = assetScale
+    this.connected = true
+    this.pool = new ServerConnectionPool(this.serverSecret, {
+      ...this.connectionOpts,
+      isServer: true,
```

Comment:
isServer: true to ServerConnectionPool seems a little redundant :))

Reply:
It is, but those options get passed through to the `Connection` constructor. Its easier to just include the flag here instead of creating another options type without it only to add it immediately.